app/backend/lib/headers.ts

```diff
@@ -0,0 +1,36 @@
+import helmet from "helmet";
+import config from "../../config";
+
+const headersMiddleware = () => {
+  const helmetMiddleware = helmet({
+    contentSecurityPolicy: false,
+  });
+  return (req, res, next) => {
+    // Tell search + crawlers not to index non-production environments:
+    if (
+      !config.get("OPENSHIFT_APP_NAMESPACE") ||
+      !config.get("OPENSHIFT_APP_NAMESPACE").endsWith("-prod")
+    ) {
+      res.append("X-Robots-Tag", "noindex, noimageindex, nofollow, noarchive");
+    }
+
+    // https://www.zaproxy.org/docs/alerts/10020-1/
+    res.append("X-Frame-Options","SAMEORIGIN");       
+
+    // https://www.zaproxy.org/docs/alerts/10021/
+    res.append("X-Content-Type-Options","nosniff");
+
+    // https://www.zaproxy.org/docs/alerts/10038/
+    res.append("Content-Security-Policy","default-src 'self'; script-src 'self'; connect-src 'self'; img-src 'self'; style-src 'self'; frame-ancestors 'self'; form-action 'self'");            
+
+    // https://www.zaproxy.org/docs/alerts/10063-1/
+    res.append("Permissions-Policy","display-capture 'none'");
+
+    // https://www.zaproxy.org/docs/alerts/10037/
+    res.removeHeader("X-Powered-By");                           
```

Comment:
Those are (or can be) set by `helmet`, no?

Reply:
```
    contentSecurityPolicy: {
      useDefaults: false,
      directives: {
        "default-src": helmet.contentSecurityPolicy.dangerouslyDisableDefaultSrc,
        "script-src": ["'self'"],
      },
    },
```

mmmh, this doesn't look more readable than what you wrote. 

I'd say if they can be enabled with helmet with a simple boolean flag (e.g. `hidePoweredBy: true`), or if they are in the helmet defaults (nosniff and X-Frame-Options are enabled by default), then let's use helmet, otherwise `res.append` is better

Reply:
Ok.. let me remove res.append entries that match default Helmet settings and test

Reply:
done